backend/Models/user.model.js

```diff
@@ -1,38 +1,98 @@
-const mongoose = require('mongoose');
+const mongoose = require("mongoose");
 
 const Schema = mongoose.Schema;
 
 // Building user schema and defining the fields
-const userSchema = new Schema({
+const userSchema = new Schema(
+  {
     userID: {
-        type: String,
-        unique: true,
-        required: true,
-        trim: true
+      type: String,
+      unique: true,
+      required: true,
+      trim: true,
     },
     refreshToken: {
-        type: String,
-        required: true,
+      type: String,
+      required: true,
     },
     name: {
-        type: String,
+      type: String,
     },
     imageURL: {
-        type: String
+      type: String,
     },
     email: {
-        type: String
+      type: String,
     },
-    groups: [
-        {
-            type: String
+
+    topTracks: [
+      {
+        songName: {
+          type: String,
+          required: true,
+        },
+        songID: {
+          type: String,
+          required: true,
+        },
+        songPopularity: {
+          type: Number,
+          required: true,
+        },
+        imageURL: {
+          type: String,
+          required: true,
+        },
+        linkURL: {
```

Comment:
Also unsure if linkURL is necssary as all songs are:
"https://open.spotify.com/track/" + trackID 

so if necessary we implicitly have it. Plus unsure of the use case unless a user can click on a song from their top tracks which does kinda make sense I guess. (additional feature though?)

Reply:
Probably easier for frontend to write readable code to leave it in